src/Transports.AspNetCore/Extensions/GraphQLBuilderUserContextExtensions.cs

```diff
@@ -62,14 +66,15 @@ public static IGraphQLBuilder AddUserContextBuilder<TUserContext>(this IGraphQLB
     /// <param name="creator">A delegate used to create the user context from the <see cref="HttpContext"/>.</param>
     /// <returns>The GraphQL builder.</returns>
     public static IGraphQLBuilder AddUserContextBuilder<TUserContext>(this IGraphQLBuilder builder, Func<HttpContext, Task<TUserContext>> creator)
-        where TUserContext : class, IDictionary<string, object>
+        where TUserContext : class, IDictionary<string, object?>
     {
-        builder.Services.Register<IUserContextBuilder>(new UserContextBuilder<TUserContext>(creator));
+        builder.Services.Register<IUserContextBuilder>(new UserContextBuilder<TUserContext>(context => new(creator(context))));
```

Comment:
+ 1 overload with `Func<HttpContext, ValueTask<TUserContext>> creator` ?

Reply:
No.  As with the entire rest of GraphQL.NET, overloads are available as synchronous or Task-based asynchronous code.  Under the hood this is enabled by using ValueTask.  Users can implement the interface directly if needed for ValueTask support.  Further, adding such an overload is a problem because the compiler will not / cannot pick between the two for an inline delegate, and so both become cumbersome to use.

Reply:
Because of the compiler issue, in our transition to support ValueTask, I have made sure that all overloads throughout the API support either a Task or ValueTask based delegate, but never both.  Generally it's set up so infrastructure like FuncFieldResolver supports ValueTask, whereas builder methods and user code (`FieldAsync` for example) supports Task.

Reply:
OK